test/extension/numeric/convolve.cpp

```diff
@@ -110,12 +110,244 @@ struct test_image_5x5_kernel_3x3_identity
     }
 };
 
+// Convention used for naming variables : 
+// 1. img : Original image which is to be used for testing.
+// 2. img_out : Output image obtained after applying 1D row and column convolution using library
+// function.
+// 3. img_expected_row : Expected image after applying row convolution on "img".
+// 4. img_expected_col : Expected image after applying column convolution on "img_expected_row".
+
+struct test_image_5x5_kernel_1x9_boundary_extend_zero
+{
+    template <typename Image>
+    void operator()(Image const&)
+    {
+        using image_t = Image;
+        using pixel_t = typename image_t::value_type;
+        using channel_t = typename gil::channel_type<pixel_t>::type;
+        auto img = fixture::generate_image<image_t>(5, 5, fixture::random_value<channel_t>{});
+        auto img_view = gil::view(img);
+        image_t img_out(img), img_expected_row(img_view.width(), img_view.height());
+        image_t img_expected_col(img_view.width(), img_view.height());
+
+        for (std::ptrdiff_t y = 0; y < img_view.height(); ++y)
+        {
+            auto img_it = img_view.row_begin(y);
+            auto img_expected_row_it = gil::view(img_expected_row).row_begin(y);
+            for (std::ptrdiff_t x = 2; x < img_view.width(); ++x)
+                img_expected_row_it[x] = img_it[x - 2];
+        }
+        for (std::ptrdiff_t x = 0; x < img_view.width(); ++x)
+        {
+            auto img_expected_row_it = gil::view(img_expected_row).col_begin(x);
+            auto img_expected_col_it = gil::view(img_expected_col).col_begin(x);
+            for (std::ptrdiff_t y = 2; y < img_view.height(); ++y)
+                img_expected_col_it[y] = img_expected_row_it[y - 2];
+        }
+
+        auto const kernel = fixture::create_kernel<channel_t>({0, 0, 0, 0, 0, 0, 1, 0, 0});
+        gil::detail::convolve_1d<pixel_t>(gil::const_view(img_out), kernel, gil::view(img_out),
```

Comment:
Shouldn't `img_out` be just `img`? I guess it got copypasted everywhere.

Reply:
Yes you are right, we can change `img_out` to `img` there. I simply chose not to because [code](https://github.com/boostorg/gil/blob/7dd61209a81695600d773f481bb8b22b0f74a30a/test/extension/numeric/convolve.cpp#L102) committed to this file earlier was using it in that manner. However, I have no objections in changing it if we are fine with that change in existing style. Shall I change it then?

Reply:
Yes, it is more natural that way and will not confuse anyone.